redhat-cop/casl-ansible
# Update CASL to use nsupdate for DNS records

#### What does this PR do?
Updates the CASL implementation to use `nsupdate` to populate DNS records in the DNS server(s). This works with both the internally provisioned DNS server as well as external DNS server(s). 

#### How should this be manually tested?
3 Test Scenarios:

1. Run the casl-ansible as-is with a valid inventory (note the updated inventory samples) and validate that the internally provisioned DNS server works as it use to 

2. Provision a "public" DNS server to be used for external access to the cluster, and leave the inventory as-is for using the private / internally provisioned DNS server within the cluster. Set the `external_nsupdate_keys['public']` configuration in the inventory to allow the tools to update an external DNS server, then run the `casl-ansible` tooling/provisioning. Observe that the cluster functions properly and that external access can be gained using the external DNS server resolution.

3. Provision 2 external DNS servers ("public" & "private"), Configure the `external_nsupdate_keys['public']` and `external_nsupdate_keys['private']` settings in the inventory, then run the `casl-ansible` tooling/provisioning. Observe that the cluster uses the "internal" DNS server for internal resolution (private IPs) while the "external" DNS server is used for public resolution. 

#### Is there a relevant Issue open for this?
resolves #8 

#### Who would you like to review this?
cc: @redhat-cop/casl @tomassedovic @cooktheryan
